File: CinemaNow/CinemaNow.Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CinemaNow.Models
{
    public class Reservation
    {
        public int Id { get; set; }
        public int? UserId { get; set; }
        public int? ScreeningId { get; set; }
        public DateTime? DateTime { get; set; }
        public int? NumberOfTickets { get; set; }
        public decimal? TotalPrice { get; set; }
        public int? PaymentId { get; set; }
        public string? PaymentType { get; set; }
        public List<ReservationSeat>? Seats { get; set; } = new List<ReservationSeat>();
        public User? User { get; set; }
        public Screening? Screening { get; set; }
        public Payment? Payment { get; set; }
        public string? QRCodeBase64 { get; set; }
    }
}

File: CinemaNow/UI/cinemanow_desktop/lib/models/reservation.dart
import 'package:cinemanow_desktop/models/reservation_seat.dart';
import 'package:cinemanow_desktop/models/screening.dart';
import 'package:cinemanow_desktop/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? id;
  int? userId;
  int? screeningId;
  DateTime? dateTime;
  int? numberOfTickets;
  double? totalPrice;
  String? paymentType;
  List<ReservationSeat>? seats;
  User? user;
  Screening? screening;

  Reservation({
    this.id,
    this.userId,
    this.screeningId,
    this.dateTime,
    this.numberOfTickets,
    this.totalPrice,
    this.paymentType,
    this.seats,
    this.user,
    this.screening,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
